thor/orbit.py

```diff
@@ -1,361 +1,299 @@
+    CometaryCoordinates,
+    CoordinateCovariances,
+    KeplerianCoordinates,
+    Origin,
+    OriginCodes,
+    SphericalCoordinates,
+    transform_coordinates,
+)
+from adam_core.observations.detections import PointSourceDetections
+from adam_core.observations.exposures import Exposures
+from adam_core.observers import Observers
+from adam_core.orbits import Ephemeris, Orbits
+from adam_core.propagator import PYOORB, Propagator
+from astropy.time import Time
+
+CoordinateType = TypeVar(
+    "CoordinateType",
+    bound=Union[
+        CartesianCoordinates,
+        SphericalCoordinates,
+        KeplerianCoordinates,
+        CometaryCoordinates,
+    ],
+)
+
+
+class RangedPointSourceDetections(qv.Table):
+
+    id = qv.StringColumn()
+    exposure_id = qv.StringColumn()
+    coordinates = SphericalCoordinates.as_column()
 
 
 class TestOrbit:
-    """
-    TestOrbit: Class that calculates and stores the rotation matrices
-    for a guess of heliocentric distance and velocity. To be used in
-    tandem with the Cell class.
+    def __init__(
+        self,
+        coordinates: CoordinateType,
+        orbit_id: Optional[str] = None,
+        object_id: Optional[str] = None,
+    ):
+        """
+        Create a test orbit from a set of orbital elements.
 
-    Parameters
-    ----------
-    cartesian : `~numpy.ndarray` (1, 6)
-        Cartesian ecliptic orbital elements with postions in units of au
-        and velocities in units of au per day.
-    t0 : `~astropy.time.core.Time` (1)
-        Epoch at which orbital elements are defined.
-    """
+        Parameters
+        ----------
+        coordinates :  `~adam_core.coordinates.cartesian.CartesianCoordinates`,
+                       `~adam_core.coordinates.spherical.SphericalCoordinates`,
+                       `~adam_core.coordinates.keplerian.KeplerianCoordinates`,
+                       `~adam_core.coordinates.cometary.CometaryCoordinates`
+            The orbital elements that define this test orbit. Can be any representation but will
+            be stored internally as Cartesian elements.
+        orbit_id : str, optional
+            Orbit ID. If not provided, a random UUID will be generated.
+        object_id : str, optional
+            Object ID, if it exists.
+        """
+        # Test orbits should be singletons
+        assert len(coordinates) == 1
+
+        # Test orbit selection will likely occur in a non-Cartesian coordinate system
+        # so we should accept any coordinate system and convert to Cartesian as the
+        # most stable representation
+        if not isinstance(coordinates, CartesianCoordinates):
+            cartesian_coordinates = coordinates.to_cartesian()
+        else:
+            cartesian_coordinates = coordinates
+
+        if orbit_id is not None:
+            self.orbit_id = orbit_id
+        else:
+            self.orbit_id = uuid.uuid4().hex
+
+        if object_id is not None:
+            self.object_id = object_id
```

Comment:
```suggestion
        self.object_id = object_id
```

Reply:
As you find more errors keep submitting these suggestions and then I'll incorporate them in a rebase because I'm crazy.

Reply:
This is fixed in #108.